### DISCUSSION TITLE
BRouter (offline) calculate long route with many waypoints fails

### DISCUSSION THREAD

**Discussion by umpfmpf** [devemux86/cruiser#438] [Q&A] [answered]
I have a long route "RouteDeGrandAlps" with a length of about 700km. 
If I import the route and calculate it offline with Brouter, I get the error message: no route found. 
In the motorcycle profile settings, I have already deactivated all options and also tried the profile car: no success. 
When I use brouters online, he calculates the route in a matter of seconds!!! 
[RouteDeGrandAlps Thonon -Menton (691km).zip](https://github.com/devemux86/cruiser/files/14212612/RouteDeGrandAlps.Thonon.-Menton.691km.zip)
To try it out, I attached the file.

**Comment by devemux86** [comment]
@umpfmpf thanks for the report.

Online routing services calculate routes on powerful desktop servers,
so it is quite different from mobile phone calculations.

I can reproduce it and will check what happens.

In any case a long route can be easily calculated offline, so it may be related to the waypoints location:

<img src="https://github.com/devemux86/cruiser/assets/3484020/5ad03b86-c6cf-4761-8fd0-5cc65e0d2fbe" width="50%"/>

**Comment by ramondevic** [comment]
I think it is better to prepare a trip of thousands of kms and do the route calculations from a computer where it will also be easier to adjust the route than to do everything from the phone.

For me calculating from the phone is just to do a little extra route to change/extend the previously prepared route calmly before starting the trip.

For me a trip is: prepare the route, execute the trip, document the trip once it's over.
What and how do you do it?

**Reply by MotoUKRider** [reply]
@ramondevic My workflow is to use "daily" route files of a sensible time/distance, planned on the PC using BaseCamp.  For a longer trip (like a loop around the coast of Scotland) I made a big looping route and added various places to visit along the way, then I chopped the route into smaller "daily" rides once I'd found a B&B/lodging nearby.

There is the possibility of the daily routes becoming out-of-sync if you have to change your plans (ie: road diversion or weather) leaving you needing to ride the last 35% of Day3.gpx and then load+ride a bit of Day4.gpx, but I've found that if you research the route and set yourself sensible goals - you can operate on the 1-Day/1-Route principle.  With Cruiser's ability to display "Next Waypoint" *and* "Route Remaining" distances/times, it also makes it easier to know when your next stop will be and how long until you finish for the day.

1 big route works, sure, but as you say it's a bit much to load+calculate it, especially if you're only working on, say, a 200-mile segment for that particular day.

I think either way here has both pros and cons :confused: 

Unless it's possible to re-calculate only the portion of the route that is changed, so if you don't change an "Avoidance" setting, then only the Shaping/Way-Points affected will need to be re-calculated.  I wonder if that's possible?

**Reply by WolfgangZ42** [reply]
> to re-calculate only the portion of the route that is changed

Would be great - and would save lot of time, preparing especially day slots of a gravel route. I think it's already planned - see  https://github.com/devemux86/cruiser/discussions/142

**Reply by umpfmpf** [reply]
For me it is important to see a route as a whole and then gradually disassemble it into individual parts and optimize it. 
Also a trip to Norway with a start and an end point, which I then split into a multi-day tour. 
In this case in the Alps, I loaded the route from the network and unfortunately I cannot view it.

**Reply by WolfgangZ42** [reply]
try this one: 
[RouteDeGrandAlps.zip](https://github.com/devemux86/cruiser/files/14217767/RouteDeGrandAlps.zip)
It's your track, imported into kurviger.de, exported as *.kurviger, imported into Cruiser and than exported as *.gpx ... but the latter one doesn't work either. Only the *.kurviger ...
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/54d86d78-6d73-439a-ab6e-bfd9c447c825"/>

**Reply by umpfmpf** [reply]
This route works also not correct: when I import the route, the calculation goes up to a number of 9 waypoints, from 10 up there is no calculation possible. 
However, the route is totally alienated with only 9 waypoints.
But when I open the Route in Kurviger and export it as *.kurviger it works fine.
However, it is not possible to import a *.kurviger file as a favorite and then calculate it from the favorites. It works with normal import.

**Reply by devemux86** [reply]
@umpfmpf 

> it is not possible to import a *.kurviger file as a favorite and then calculate it from the favorites. It works with normal import.

You can store kurviger routes as favorites in Cruiser with groups and display them with different colors:
- Select "Favorites | Routes"
- Import your routes
- Then can load each favorite route and make changes

<img src="https://github.com/devemux86/cruiser/assets/3484020/73e8f2e3-ec0a-455c-a87d-56d3c4582dd1" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/3066a2d7-288d-4559-805d-aa62af027f10" width="50%"/>

**Reply by umpfmpf** [reply]
I had put it wrong: I can import the *.kurviger route as a favorite, but he does not calculate it and only shows the waypoints without an error message.

**Reply by 0709wiwiwi** [reply]
[RouteDeGrandAlps_navtrk.gpx.txt](https://github.com/devemux86/cruiser/files/14219374/RouteDeGrandAlps_navtrk.gpx.txt)

**Reply by umpfmpf** [reply]
It seems like it works.
What have you done?

**Comment by devemux86** [comment]
@umpfmpf thanks very much for your report! :+1: 

Offline calculation of such long routes with too many waypoints will work properly in the next Cruiser update.

It made me review BRouter (offline) algorithms and now Cruiser will use a new simpler and faster workflow. :rocket: 

---

The improvement will be available in Cruiser's **internal** BRouter (offline) implementation.
(enabled by default)

Calling the external BRouter app uses its own separate workflow and requires extra work.
cc @afischerdev

<img src="https://github.com/devemux86/cruiser/assets/3484020/20d6243a-2b21-447f-9213-5380dcb557d4" width="50%"/>

**Reply by devemux86** [reply]
> Calling the external BRouter app uses its own separate workflow and requires extra work.

It will also work with the external BRouter app.

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.40](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
